From f1395797653664e67e778aa7f7aca01421f90e94 Mon Sep 17 00:00:00 2001
From: Tim Lunn <tim@feathertop.org>
Date: Thu, 14 Feb 2013 09:52:30 +1100
Subject: [PATCH] Only install the REAL_LIBRARY

---
 js/src/Makefile.in | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/js/src/Makefile.in b/js/src/Makefile.in
index 28a1e82..96ef694 100644
--- a/js/src/Makefile.in
+++ b/js/src/Makefile.in
@@ -867,9 +867,9 @@ install:: $(INSTALLED_HEADERS)
 install:: $(SCRIPTS)
 	$(SYSINSTALL) $^ $(DESTDIR)$(bindir)
 
-install:: $(LIBRARY) $(SHARED_LIBRARY) $(IMPORT_LIBRARY)
-ifneq (,$(LIBRARY))
-	$(SYSINSTALL) $(LIBRARY) $(DESTDIR)$(libdir)
+install:: $(REAL_LIBRARY) $(SHARED_LIBRARY) $(IMPORT_LIBRARY)
+ifneq (,$(REAL_LIBRARY))
+	$(SYSINSTALL) $(REAL_LIBRARY) $(DESTDIR)$(libdir)
 endif
 ifneq (,$(SHARED_LIBRARY))
 	$(SYSINSTALL) $(SHARED_LIBRARY) $(DESTDIR)$(libdir)
-- 
1.8.1.2


