From 040fa32fb4b193e8c8ae2e2bf3c5a8e4841817dc Mon Sep 17 00:00:00 2001
From: Tim Lunn <tim@feathertop.org>
Date: Mon, 11 Feb 2013 11:33:06 +1100
Subject: [PATCH 14/24] fix pkgconfig file

---
 js/src/js.pc.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/src/js.pc.in b/js/src/js.pc.in
index a6385bd..4839791 100644
--- a/js/src/js.pc.in
+++ b/js/src/js.pc.in
@@ -8,4 +8,4 @@ Description: The Mozilla library for JavaScript
 Version: @MOZILLA_VERSION@
 Requires: @NSPR_PKGCONF_CHECK@
 Libs: -L${libdir} -l@LIBRARY_NAME@
-Cflags: -I${includedir}/js/RequiredDefines.h -I${includedir}/@LIBRARY_NAME@
+Cflags: -include ${includedir}/@LIBRARY_NAME@/js/RequiredDefines.h -I${includedir}/@LIBRARY_NAME@
-- 
1.8.1.2

