diff --git a/js/src/Makefile.in b/js/src/Makefile.in
index 2d3ee7e..03e2a45 100755
--- a/js/src/Makefile.in
+++ b/js/src/Makefile.in
@@ -33,8 +33,15 @@ endif
 
 TEST_DIRS += tests
 
+SRCREL_VERSION      = 1.0.0
+SRCREL_ABI_VERSION  := $(word 1,$(subst ., ,$(SRCREL_VERSION))).$(word 2,$(subst ., ,$(SRCREL_VERSION)))
+
 MODULE		    = js
-LIBRARY_NAME	    = mozjs
+ifeq (,$(HOST_BIN_SUFFIX))
+LIBRARY_NAME	    = mozjs188
+else
+LIBRARY_NAME	    = mozjs188-$(SRCREL_ABI_VERSION)
+endif
 STATIC_LIBRARY_NAME = js_static
 GRE_MODULE	    = 1
 
@@ -726,6 +733,7 @@ endif
 # - OS_LIBS includes libraries selected by the configure script.
 # - EXTRA_LIBS includes libraries selected by this Makefile.
 JS_CONFIG_LIBS=$(EXTRA_DSO_LDOPTS) $(OS_LIBS) $(EXTRA_LIBS)
+JS_CONFIG_MOZ_JS_LIBS=-L$(libdir) -l$(LIBRARY_NAME)
 
 # The configure script invokes this rule explicitly at configure time!
 # It's important that js-config be ready by the time we're done
@@ -751,24 +759,70 @@ js-config: js-config.in Makefile $(DEPTH)/config/autoconf.mk $(topsrcdir)/config
 	-e 's|$(at)LIBRARY_NAME$(at)|$(LIBRARY_NAME)|' \
 	-e 's|$(at)NSPR_CFLAGS$(at)|$(NSPR_CFLAGS)|' \
 	-e 's|$(at)JS_CONFIG_LIBS$(at)|$(JS_CONFIG_LIBS)|' \
-	-e 's|$(at)MOZ_JS_LIBS$(at)|$(MOZ_JS_LIBS)|' \
+	-e 's|$(at)MOZ_JS_LIBS$(at)|$(JS_CONFIG_MOZ_JS_LIBS)|' \
 	&& mv js-config.tmp $@ && chmod +x $@
 
 SCRIPTS = js-config
 SDK_BINARY = js-config
 
+pkg_config_files = \
+	mozjs188.pc \
+	$(NULL)
+
+GARBAGE += $(pkg_config_files)
+
+%.pc: $(srcdir)/%.pc.in Makefile
+	cat $< | sed \
+	-e "s|%prefix%|$(prefix)|" \
+	-e "s|%includedir%|$(includedir)|" \
+	-e "s|%libdir%|$(libdir)|" \
+	-e 's|%MOZILLA_VERSION%|$(MOZILLA_VERSION)|' \
+	> $@ && chmod 644 $@
+
+install:: $(pkg_config_files)
+	@echo pkg_config_file: $(pkg_config_files)
+	$(SYSINSTALL) $(IFLAGS1) $^ $(DESTDIR)$(libdir)/pkgconfig
+
 install:: $(INSTALLED_HEADERS)
 	$(SYSINSTALL) $^ $(DESTDIR)$(includedir)/$(MODULE)
 
 install:: $(SCRIPTS)
 	$(SYSINSTALL) $^ $(DESTDIR)$(bindir)
 
+ifeq (,$(HOST_BIN_SUFFIX))
+ifeq (.dylib,$(DLL_SUFFIX))
+# Mac OS X
+SHLIB_ANY_VER   := $(DESTDIR)$(libdir)/$(SHARED_LIBRARY)
+SHLIB_ABI_VER   := $(DESTDIR)$(libdir)/$(SHARED_LIBRARY:$(DLL_SUFFIX)=.$(SRCREL_ABI_VERSION)$(DLL_SUFFIX))
+SHLIB_EXACT_VER := $(DESTDIR)$(libdir)/$(SHARED_LIBRARY:$(DLL_SUFFIX)=.$(SRCREL_VERSION)$(DLL_SUFFIX))
+$(SHARED_LIBRARY): EXTRA_DSO_LDOPTS += -install_name $(SHLIB_ABI_VER)
+$(SHARED_LIBRARY): EXTRA_DSO_LDOPTS += -current_version $(SRCREL_VERSION)
+$(SHARED_LIBRARY): EXTRA_DSO_LDOPTS += -compatibility_version $(SRCREL_ABI_VERSION)
+else
+# Generic Unix / Linux
+SHLIB_ANY_VER   := $(DESTDIR)$(libdir)/$(SHARED_LIBRARY)
+SHLIB_ABI_VER   := $(DESTDIR)$(libdir)/$(SHARED_LIBRARY).$(SRCREL_ABI_VERSION)
+SHLIB_EXACT_VER := $(DESTDIR)$(libdir)/$(SHARED_LIBRARY).$(SRCREL_VERSION)
+$(SHARED_LIBRARY): EXTRA_DSO_LDOPTS += -Wl,-soname,$(notdir $(SHLIB_ABI_VER))
+endif
+endif
+
 install:: $(LIBRARY) $(SHARED_LIBRARY) $(IMPORT_LIBRARY)
 ifneq (,$(LIBRARY))
 	$(SYSINSTALL) $(LIBRARY) $(DESTDIR)$(libdir)
+	mv -f $(DESTDIR)$(libdir)/$(LIBRARY) $(subst $(STATIC_LIBRARY_NAME),mozjs188-$(SRCREL_ABI_VERSION),$(DESTDIR)$(libdir)/$(LIBRARY))
 endif
 ifneq (,$(SHARED_LIBRARY))
+ifeq (,$(HOST_BIN_SUFFIX))
+	@[ ! -h $(SHLIB_ANY_VER) ] || rm -f $(SHLIB_ANY_VER)
+endif
 	$(SYSINSTALL) $(SHARED_LIBRARY) $(DESTDIR)$(libdir)
+ifeq (,$(HOST_BIN_SUFFIX))
+	mv -f $(SHLIB_ANY_VER) $(SHLIB_EXACT_VER)
+	@[ ! -h $(SHLIB_ABI_VER) ] || rm -f $(SHLIB_ABI_VER)
+	ln -s $(notdir $(SHLIB_EXACT_VER)) $(SHLIB_ABI_VER)
+	ln -s $(notdir $(SHLIB_ABI_VER)) $(SHLIB_ANY_VER)
+endif
 endif
 ifneq (,$(IMPORT_LIBRARY))
 	$(SYSINSTALL) $(IMPORT_LIBRARY) $(DESTDIR)$(libdir)
diff --git a/js/src/mozjs188.pc.in b/js/src/mozjs188.pc.in
new file mode 100644
index 0000000..a309aa4
--- /dev/null
+++ b/js/src/mozjs188.pc.in
@@ -0,0 +1,10 @@
+prefix=%prefix%
+libdir=%libdir%
+includedir=%includedir%
+
+Name: SpiderMonkey 1.8.8
+Description: The Mozilla library for JavaScript 1.8.8
+Version: %MOZILLA_VERSION%
+Requires: nspr >= 4.9.2
+Libs: -L${libdir} -lmozjs188
+Cflags: -I${includedir}/js
